pkg/bootstrap/platform/gcp.go

```diff
@@ -128,14 +129,29 @@ func (e *gcpEnv) Metadata() map[string]string {
 	return md
 }
 
+var (
+	once        sync.Once
+	envPid      string
+	envNpid     string
+	envCluster  string
+	envLocation string
+)
+
 func parseGCPMetadata() (pid, npid, cluster, location string) {
-	gcpmd := gcpMetadataVar.Get()
-	log.Infof("Extract GCP metadata from env variable GCP_METADATA: %v", gcpmd)
-	parts := strings.Split(gcpmd, "|")
-	if len(parts) != 4 {
-		return
-	}
-	return parts[0], parts[1], parts[2], parts[3]
+	once.Do(func() {
```

Comment:
I would really prefer if we didn't need to do once.Do and could just restructure the code instead of using heavyweight features like this..